avm/src/vm/opcode.rs

```diff
@@ -2682,12 +2675,11 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
         hm.register_out(args[0], i, CLOSURE_ARG_MEM_START + 1);
         filters.push(subhandler.clone().run(hm));
       }
-      let mut hms = join_all(filters).await;
+      let hms = join_all(filters).await;
       let mut idxs = vec![];
-      for i in 0..len {
-        let hm = &mut hms[i];
+      for (i, hm) in hms.into_iter().enumerate() {
         let val = hm.read_fixed(CLOSURE_ARG_MEM_START);
-        hm.drop_parent();
+        hm.drop_parent(); // this drops `hm`
```

Comment:
Wait, what? Why does that happen? Why can't we just call `drop(hm);` if that's the case?

Reply:
https://github.com/alantech/alan/pull/391#discussion_r564812823

Reply:
oh also, `hm` is dropped on line 2686 anyways, so it doesn't actually impact the loop, I just left this here so people don't get confused if they can't access `hm` on the next line or something

Reply:
Is that because it now returns `hm`, so if you don't grab its return value its effectively dropped?

It may make sense to rename these two functions, since it doesn't just drop the parent reference anymore?

Reply:
> Is that because it now returns `hm`, so if you don't grab its return value its effectively dropped?

Yes

> It may make sense to rename these two functions, since it doesn't just drop the parent reference anymore?

I mean, it still does. Effectively, the only difference between `drop_parent` and `Drop::<HandlerMemory>::drop` is that `drop_parent` *only* drops the parent, and still allows you to access the `HandlerMemory` afterwards.